### PR TITLE
DCS-1239 Add 3 additional forms and tidy up date handling

### PR DESCRIPTION
**Pull request by andrewrlee**
Also adding in Extremism conditions and reordering

### CODE REVIEW

File: server/services/utils/conditionsFormatter.js [hunk 1]
```diff
@@ -1,66 +0,0 @@
...
-
-function getSelectedFieldNamesReducer(array, condition) {
-  if (!condition.field_position) {
-    return array
-  }
-  const inputItems = Object.keys(condition.field_position).map((key) => key)
-  return [...array, ...inputItems]
-}
-
-function combineDatesIn(conditionsFieldsRequired, inputObject) {
```
Comment (andrewrlee) [inline server/services/utils/conditionsFormatter.js:45]: rewrote this function to make it data driven and added a new type of date field for trial monitoring

File: server/services/utils/conditionsFormatter.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      const year = inputObject[values.year]
+      if (day && month && year) {
+        acc[name] = `${day}/${month}/${year}`
+      }
+    }
+    return acc
+  }, {})
+
+  return merge(inputObject, combinedDates)
+}
```
Comment (andrewrlee) [inline server/services/utils/conditionsFormatter.ts:77]: typescripted this file and rewrote this function ^^

File: server/services/conditionsService.ts [hunk 3]
```diff
@@ -187,41 +186,24 @@ function splitIntoGroupedObject(conditionObject, condition) {
...
-function getSubmissionForCondition(conditionId, inputtedConditions: AdditionalConditions) {
-  if (isEmpty(inputtedConditions[conditionId])) {
+function getSubmissionForCondition(condition: ConditionMetadata, inputtedConditions: AdditionalConditions) {
+  const { id } = condition
+  const inputtedCondition = inputtedConditions[id]
+  if (isEmpty(inputtedCondition)) {
     return {}
   }
 
-  if (conditionId === 'ATTENDDEPENDENCY') {
```
Comment (andrewrlee) [inline server/services/conditionsService.ts:203]: pushed this logic into `displayForEdit` in the condition to make this function more generic and removed the requirement to have specific logic for individual conditions

**Review by louisenorris — APPROVED** [review]
LGTM

**Pull request merged**